Edited File Format and File Path entries (#1270)
Codecademy/docs: content/general/concepts/file-paths/file-paths.md

- Files that may link to programs, such as [CDNs](https://www.codecademy.com/resources/docs/general/cdn), [frameworks](https://www.codecademy.com/resources/docs/general/cdn), or [Node applications](https://www.codecademy.com/resources/docs/open-source/node-js)
- Other files, such as [JSON](https://www.codecademy.com/resources/docs/general/json) or plain text

## URLs and File Paths

[URLs](https://www.codecademy.com/resources/docs/general/url) specify the location where data and websites are hosted on the web. URLs and file paths both provide the location a computer must go to to find a particular resource. A major difference between URLs and file paths is that URLs locate resources which are hosted on the Internet, and file paths locate resources which are stored locally or on a local network.

## Absolute and Relative File Paths

Absolute file paths specify the location of a file from the root directory in the file system structure. They are also called "full file paths" or "full paths". Some examples of absolute file paths are:

- `/home/user/website/main/about.html`
- `/home/user/projects/js/script.js`
- `/home/user/data-analysis/scripts/main.py`

> In Linux, the tilde (~) is commonly used to represent a user's home directory in a file path. For example, the above file paths could be changed to:

Reviewer: ```suggestion
In [Linux](https://www.codecademy.com/resources/docs/open-source/linux), the tilde (~) is commonly used to represent a user's home directory in a file path. For example, the above file paths could be changed to:
```

- `~/website/main/about.html`
- `~/projects/js/script.js`
- `~/data-analysis/scripts/main.py`

Relative file paths specify the location of a file in the same folder or on the same server. In other words, a relative file path specifies a location of a file that is relative to the currect directory. Some examples of relative file paths are:

- `./about.html`
- `./js/script.js`
- `./scripts/main.py`

> Relative file paths use a dot notation at the start of the path, followed by a path separator and the location of the file. A single dot (.) indicates the current directory, and a double dot (..) indicates the parent directory. In [Unix](https://www.codecademy.com/resources/docs/general/unix)-based operating systems, such as [Linux](https://www.codecademy.com/resources/docs/open-source/linux) or macOS, `./` tells the program to look for the file from the current directory. `../` tells the program to go up to the parent directory before looking for the file. In Windows, the path separator is a back slash (\\), so `.\` and `..\` are used instead.

Reviewer: ```suggestion
Relative file paths use a dot notation at the start of the path, followed by a path separator and the location of the file. A single dot (.) indicates the current directory, and a double dot (..) indicates the parent directory. In [Unix](https://www.codecademy.com/resources/docs/general/unix)-based operating systems, such as Linux or macOS, `./` tells the program to look for the file from the current directory. `../` tells the program to go up to the parent directory before looking for the file. In Windows, the path separator is a back slash (\\), so `.\` and `..\` are used instead.
```

> In [Unix](https://www.codecademy.com/resources/docs/general/unix)-based operating systems, such as Linux or macOS, `./` tells the program to look for the file from the current directory. `../` tells the program to go up to the parent directory before looking for the file. In Windows, the path separator is a back slash (\\), so `.\` and `..\` are used instead.

I feel like this chunk of text is better suited with the information on line # 25 (in the "URLs and File Paths" section). Based on [this IBM source I came across](https://www.ibm.com/docs/en/zvse/6.2?topic=SSB27H_6.2.0/fa2ad_use_forward_or_backward_slashes_under_windows.htm), should we include a note (**on this line**) about how forward and back slashes can be used in Windows, but that paths with back slashes might be confused as relative instead of absolute?

Author: I think it might be confusing if this section was moved up to the URL & File Paths section, since URLs follow a standard format that always use forward slash regardless of the OS.

Would it be better to just change the structure of this entry? So it would be:
- Intro to file paths information 
  - Keep same first section the same
- Paths and Path Separators
  - Expand on some general information about paths and how they work (e.g., describe how they relate to the file system structure with directories separated by path separators)
  - Put the "In Unix..." part you suggested in this section
- Absolute and Relative File Paths 
  - Cut it off where you suggested
- URLs and File Paths
  - Keep the same information, but add a bit more information that links this section to the rest of the article's content (e.g., "URLs and file paths function similarly, but...)

And/or, I could add a little more info on paths/path separators to the [URL](https://www.codecademy.com/resources/docs/general/url) page.

Regrading the Windows part, I'd be hesitant to add information about Windows being compatible with forward slashes, since I don't think that's a best practice for Windows developers, unless interacting with Unix/Linux systems. [This link from Microsoft's docs](https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats) only uses back slashes in their examples, and it says that all forward slashes are converted to back slashes. 

I could definitely reword that though, if you'd prefer it was less strong about the Windows path separators (e.g., "back slash should be used, although many Windows systems are compatible with the forward slash").

I think the IBM article is saying that their specific IBM system (VTAPE?) may misinterpret the backslash, so they recommend just using forward slashes to avoid errors, since Windows accepts them anyway. So it's just for IBM.

  - 'paths/computer-science'
---

A **file path** specifies the location of a file in a computer's [file system structure](https://www.codecademy.com/resources/docs/general/file-system-structure). They can be used to locate files and web resources, including:

- Media files, such as images
- Links to [CSS stylesheets](https://www.codecademy.com/resources/docs/css) for a web page
- Files that may link to programs, such as [CDNs](https://www.codecademy.com/resources/docs/general/cdn), [frameworks](https://www.codecademy.com/resources/docs/general/cdn), or [Node applications](https://www.codecademy.com/resources/docs/open-source/node-js)
- Other files, such as [JSON](https://www.codecademy.com/resources/docs/general/json) or plain text

## Paths and Path Separators

> In general, a path is a string of characters which specifies a unique location in a directory or page hierarchy. For file systems, each level in the hierarchy is a directory. For [URLs](https://www.codecademy.com/resources/docs/general/url), each level in the hierarchy is a page. Different sections of the path are separated by a path separator, such as a forward slash (/). These different sections represent the separate directories or pages in the hierarchy. For example, consider the following file path:

Reviewer: ```suggestion
In general, a path is a string of characters which specifies a unique location in a directory or page hierarchy. For file systems, each level in the hierarchy is a directory. For [URLs](https://www.codecademy.com/resources/docs/general/url), each level in the hierarchy is a page. Different sections of the path are separated by a path separator, such as a forward slash (/). These different sections represent the separate directories or pages in the hierarchy, as shown below:
```

For Docs entries, we'd like the information to be as manual and non-instructive as possible (i.e., "as shown below" instead of "consider the following ...").

`/home/user/python/test.py`

In this file path, the `test.py` file is inside the `python` directory. The `python` directory is a subdirectory of the `user` directory, which is a subdirectory of the `home` directory. File paths are one example of a path. Paths are also an important component of URLs.

In [Unix](https://www.codecademy.com/resources/docs/general/unix)-based operating systems, such as Linux or macOS, the path separator is a forward slash (/). In Windows, the path separator is a back slash (\\).

> **Note:** URLs follow a standard format that always uses a forward slash (/) as the path separator regardless of the operating system.

## URLs and File Paths

URLs specify the location where data and websites are hosted on the web. URLs and file paths both function similarly and provide the location a computer must go to to find a particular resource. However, a major difference is that URLs locate resources which are hosted on the Internet, and file paths locate resources which are stored locally or on a local network.

## Absolute and Relative File Paths

> Absolute file paths specify the location of a file from the root directory in the file system structure. They are also called "full file paths" or "full paths". Some examples of absolute file paths are:

Reviewer: ```suggestion
Absolute file paths specify the location of a file from the root directory in the file system structure. They are also called "full file paths" or "full paths." Some examples of absolute file paths are:
```
